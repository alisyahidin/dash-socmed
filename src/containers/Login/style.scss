.login-page {
  height: 100vh;
}