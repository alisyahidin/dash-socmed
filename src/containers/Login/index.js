import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import classNames from 'classnames'
import { Button, Form } from 'react-bootstrap'

import { login } from '../../actions/auth'
import './style.scss'

class Login extends Component {
  state = {
    username: '',
    email: '',
  }

  handleChange = ({target}) => {
    this.setState({
      [target.name]: target.value
    })
  }

  handleLogin = e => {
    e.preventDefault()
    const { login } = this.props

    login(this.state)
  }

  componentWillUnmount() {
    this.setState({ username: '', email: '' })
  }

  componentDidUpdate() {
    if (this.props.auth.user !== null) {
      this.props.history.push('/')
    }
  }

  render() {
    const { auth } = this.props
    const { username, email } = this.state

    return (
      <div className="login-page">
        <div className="h-75 d-flex flex-column align-items-center justify-content-center">
          <p
            className={classNames(
              "message text-danger mb-3",
              auth.error !== null ? 'message--show' : 'message--hide'
            )}
          >
            {auth.error}
          </p>
          <Form className="text-center w-25 px-5 d-flex flex-column justify-content-center">
            <Form.Group controlId="username">
              <Form.Control
                onChange={this.handleChange}
                name="username"
                value={username}
                type="text"
                placeholder="Username e.g: Bret"
              />
            </Form.Group>
            <Form.Group controlId="email">
              <Form.Control
                onChange={this.handleChange}
                name="email"
                value={email}
                type="email"
                placeholder="Email e.g: Sincere@april.biz"
              />
            </Form.Group>
            <Button
              onClick={this.handleLogin}
              type="submit"
              variant="primary"
              className="mt-sm-2 mx-auto"
              disabled={auth.loading}
            >
              {auth.loading ? 'Loading': 'Login'}
            </Button>
          </Form>
        </div>
        <div className="h-25 text-secondary d-flex align-items-center justify-content-center">
          Copyright &copy; 2019
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => ({ ...state })

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    login
  }, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(Login)