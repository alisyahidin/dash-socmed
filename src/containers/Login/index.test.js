import React from 'react'
import { shallow } from 'enzyme'
import Login from './index'

it('render <Login /> correctly', () => {
  const component = shallow(<Login />)

  expect(component).toMatchSnapshot()
})