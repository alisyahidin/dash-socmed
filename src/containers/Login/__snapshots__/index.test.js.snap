// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Login /> correctly 1`] = `
<div
  className="login-page"
>
  <div
    className="h-75 d-flex flex-column align-items-center justify-content-center"
  >
    <p
      className="message text-danger mb-3 message--show"
    />
    <ForwardRef(Bootstrap(Form))
      className="text-center w-25 px-5 d-flex flex-column justify-content-center"
    >
      <ForwardRef(Bootstrap(FormGroup))
        controlId="username"
      >
        <ForwardRef(ContextTransform)
          name="username"
          placeholder="Username e.g: Bret"
          type="text"
        />
      </ForwardRef(Bootstrap(FormGroup))>
      <ForwardRef(Bootstrap(FormGroup))
        controlId="email"
      >
        <ForwardRef(ContextTransform)
          name="email"
          placeholder="Email e.g: Sincere@april.biz"
          type="email"
        />
      </ForwardRef(Bootstrap(FormGroup))>
      <ForwardRef(Bootstrap(Button))
        className="mt-sm-2 mx-auto"
        type="button"
        variant="primary"
      >
        Login
      </ForwardRef(Bootstrap(Button))>
    </ForwardRef(Bootstrap(Form))>
  </div>
  <div
    className="h-25 d-flex align-items-center justify-content-center"
  >
    Copyright Â© 2019
  </div>
</div>
`;
